src/CMakeLists.txt

```diff
@@ -0,0 +1,127 @@
+# Must be included before CMAKE_INSTALL_INCLUDEDIR is used.
+include(GNUInstallDirs)
+set(${PROJECT_NAME}_installables "")
+
+if(SECP256K1_ASM STREQUAL "arm")
+  add_library(common OBJECT
+    asm/field_10x26_arm.s
+  )
+  set(common_obj "$<TARGET_OBJECTS:common>")
+else()
+  set(common_obj "")
+endif()
+
+add_library(precomputed OBJECT
+  precomputed_ecmult.c
+  precomputed_ecmult_gen.c
+)
+set(internal_obj "$<TARGET_OBJECTS:precomputed>" "${common_obj}")
+
+add_library(${PROJECT_NAME} SHARED EXCLUDE_FROM_ALL
+  secp256k1.c
+  ${internal_obj}
+)
+target_include_directories(${PROJECT_NAME} INTERFACE
+  $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+)
+target_compile_definitions(${PROJECT_NAME} PRIVATE
+  $<$<PLATFORM_ID:Windows>:DLL_EXPORT>
```

Comment:
Why only Windows?

Also, as a follow-up, we should change this define to `SECP256k1_DLL_EXPORT` or so, to get away from the libtool naming confusion once and for all.

Reply:
> Also, as a follow-up, we should change this define to `SECP256k1_DLL_EXPORT` or so, to get away from the libtool naming confusion once and for all.

added to the list